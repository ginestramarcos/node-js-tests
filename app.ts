import express from 'express';
import cookieParser from 'cookie-parser';
import createError from 'http-errors';

import logger from 'morgan';
import * as nunjucks from 'nunjucks';
import * as path from 'path';

import {indexHandler} from './routes/index';
import {userHandler} from './routes/users';

export const app = express();

// view engine setup
app.set('views', path.join(__dirname, 'views'));
app.set('view engine', 'html');

nunjucks.configure('views', {
  express: app,
  autoescape: true,
});
app.set('view engine', 'html');

app.use(logger('dev'));
app.use(cookieParser());
app.use(express.json());
app.use(express.urlencoded({extended: false}));
app.use(express.static(path.join(__dirname, 'public')));

const router: express.Router = new (express.Router as any)();
app.use('/', router.get('/', indexHandler));
app.use('/users', router.get('/users', userHandler));

// catch 404 and forward to error handler
app.use(function(req: express.Request, res: express.Response,
    next: express.NextFunction) {
  next(createError(404));
});

// error handler
app.use(function(
    err: any, req: express.Request, res: express.Response,
    next: express.NextFunction) {
  // set locals, only providing error in development
  res.locals.message = err.message;
  res.locals.error = req.app.get('env') === 'development' ? err : {};

  // render the error page
  res.status(err.status || 500);
  console.log(err + ' ' + err.stack);
  res.render('error');
});
