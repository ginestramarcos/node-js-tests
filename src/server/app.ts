import express from 'express';
import cookieParser from 'cookie-parser';
import logger from 'morgan';
import path from 'path';

import {additionHandler} from './routes/addition';

export const port = parseInt(process.env.PORT || '3000', 10);

export const app = express();
app.set('port', port);

app.use(logger('dev'));
app.use(cookieParser());
app.use(express.json());
app.use(express.urlencoded({extended: false}));

const publicPath = path.join(__dirname, '../../dist/client');
const landingFile = path.join(publicPath, 'index.html');
app.use(express.static(publicPath));

const router: express.Router = new (express.Router as any)();
router.get('/addition', additionHandler);
app.use(router);

// catch 404 and forward to error handler
app.use(function(req: express.Request, res: express.Response,
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    next: express.NextFunction) {
  res.sendFile(landingFile);
});

// error handler
app.use(function(
    err: any, req: express.Request, res: express.Response,
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    next: express.NextFunction) {
  // set locals, only providing error in development
  res.locals.message = err.message;
  res.locals.error = req.app.get('env') === 'development' ? err : {};

  // render the error page
  res.status(err.status || 500);
  console.log(err + ' ' + err.stack);
  res.render('error');
});
