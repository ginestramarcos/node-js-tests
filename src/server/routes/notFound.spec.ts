import {app} from '../app';
import supertest from 'supertest';

const request = supertest(app);

it('not found delegated to client', async (done) => {
  const response = await request.get('/blahblah');
  expect(response.status).toBe(200);
  expect(response.text).toMatchSnapshot();
  done();
});
