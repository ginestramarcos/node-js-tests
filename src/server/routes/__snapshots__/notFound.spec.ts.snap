// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`not found delegated to client 1`] = `"<!doctype html><html lang=\\"en\\"><head><meta charset=\\"UTF-8\\"><meta name=\\"viewport\\" content=\\"width=device-width,initial-scale=1,maximum-scale=1\\"><meta http-equiv=\\"X-UA-Compatible\\" content=\\"ie=edge\\"><link rel=\\"shortcut icon\\" type=\\"image/x-icon\\" href=\\"favicon.ico\\"/><title>React Boilerplate</title><link rel=\\"icon\\" href=\\"favicon.ico\\"></head><body><div id=\\"root\\"></div><script src=\\"App.js\\"></script></body></html>"`;
